haystack/nodes/answer_generator/openai.py

```diff
@@ -186,6 +195,23 @@ def __init__(
             logger.debug("Using GPT2TokenizerFast")
             self._hf_tokenizer: PreTrainedTokenizerFast = GPT2TokenizerFast.from_pretrained(tokenizer)
 
+    def using_azure(self) -> bool:
+        return self.deployment_name is not None and self.base_url is not None
```

Comment:
Actually, could we just make this a member variable in the `__init__` function? I don't think there is any reason to reevaluate this boolean each time we want to know if the Azuer API should be used since we haven't implemented the ability to allow these variables to be updated at runtime.